src/pal/inc/unixasmmacrosarm.inc

```diff
@@ -146,17 +146,46 @@ C_FUNC(\Name\()_End):
 
         __PWTB_StackAlloc = __PWTB_TransitionBlock
 
+	// Make the stack operation compatible with GDB and LIBUNWIND
+	push {r7, lr}
+	mov r7, sp
+	sub sp, #((__PWTB_StackAlloc) + 4 * 4 + 9 * 4)
+
         .ifnc \pushArgRegs, DoNotPushArgRegs
-                PUSH_ARGUMENT_REGISTERS
+		sub r7, #12
+		str r0, [r7]
+		str r1, [r7 , #4]
+		str r2, [r7 , #8]
+		str r3, [r7 , #12]
         .endif
 
-        PUSH_CALLEE_SAVED_REGISTERS
-        
-        alloc_stack     __PWTB_StackAlloc
+	sub r7, #36
+	str r4, [r7]
+	str r5, [r7 , #4]
+	str r6, [r7 , #8]
+	// Store R7 down there.
+	str r8, [r7 , #16]
+	str r9, [r7 , #20]
+	str r10, [r7 , #28]
+	str r11, [r7 , #32]
+	str lr, [r7 , #36]
+
+	// Get R7 from the first push {r7, lr}
+	ldr r6, [sp , #((__PWTB_StackAlloc) + 4 * 4 + 9 * 4 + 4)]
+	str r6, [r7 , #12]
+
+        //alloc_stack     __PWTB_StackAlloc
 
         .if (__PWTB_SaveFPArgs == 1)
-                add r6, sp, #(__PWTB_FloatArgumentRegisters)
-                vstm r6, {d0-d7}
```

Comment:
I still don't have a clue; but when this VSTM is moved before push{...lr}, GDB recognises the stack frame of ThePreStub and above from below the ThePreStub (e.g., PreStubWork). Actually, an initial draft of this patch was doing
```
VPUSH d0-d7
PUSH r0-r3
PUSH r4-r11, LR
...
```
, which allowed proper GDB stack backtracking through ThePreStub. However, that draft breaks CoreCLR as it breaks the struct given as an argument of PreStubWork().

If we were simply let the code run and not try to travel through stack frames or save/restore context, we don't need to care of using VSTM, PUSH, VPUSH, STM or whatsoever. However, as we are traveling though stack frames, saving and restoring contexts, I wanted to start by following what others (compilers) doing for formatting the function stack schemes, which is I'm doing here.

From the view of stack unwinder or debugger, what you see is the LR of the "down" frame for "up" frame. And it appears (I didn't look into GDB/Libunwind code much yet) that they try to find the next LR by searching for the closest lower address than the LR with stack pushing instruction, which requires us to format the sequence of instructions exactly as supposed. I'll need to look at what the protocol is later before making this non-WIP, it looks like this:
```
  >�ransitionBlock*, MethodDesc*)>       push      {r7, lr}                    �
   �ransitionBlock*, MethodDesc*)+2>     mov       r7, sp                      �
   �ransitionBlock*, MethodDesc*)+4>     sub.w     sp, sp, #880    ; 0x370   
```